plugin/pkg/global/resourcereferencemanager/admission_test.go

```diff
@@ -684,5 +684,239 @@ var _ = Describe("resourcereferencemanager", func() {
 				})))
 			})
 		})
+
+		Context("CloudProfile - Update Kubernetes versions", func() {
+			versions := []core.ExpirableVersion{
+				{Version: "1.17.2"},
+				{Version: "1.17.1"},
+				{Version: "1.17.0"},
+			}
+			shootOne := shootBase
+			shootOne.Name = "shoot-One"
+			shootOne.Spec.Provider.Type = "aws"
+			shootOne.Spec.Kubernetes.Version = "1.17.2"
+			shootTwo := shoot
+			shootTwo.Name = "shoot-Two"
+			shootTwo.Spec.Provider.Type = "aws"
+			shootTwo.Spec.Kubernetes.Version = "1.17.1"
+			var (
+				cloudProfile = core.CloudProfile{
+					Spec: core.CloudProfileSpec{
+						Kubernetes: core.KubernetesSettings{
+							Versions: versions,
+						},
+					},
+				}
+			)
+			It("should accept removal of kubernetes version that is not in use by any shoot", func() {
+				Expect(gardenCoreInformerFactory.Core().InternalVersion().Shoots().Informer().GetStore().Add(&shootOne)).To(Succeed())
+				Expect(gardenCoreInformerFactory.Core().InternalVersion().Shoots().Informer().GetStore().Add(&shootTwo)).To(Succeed())
+
+				cloudProfileNew := core.CloudProfile{
+					Spec: core.CloudProfileSpec{
+						Kubernetes: core.KubernetesSettings{
+							Versions: []core.ExpirableVersion{
+								{Version: "1.17.2"},
+								{Version: "1.17.1"},
+							},
+						},
+					},
+				}
+
+				attrs := admission.NewAttributesRecord(&cloudProfileNew, &cloudProfile, core.Kind("CloudProfile").WithVersion("version"), "", cloudProfile.Name, core.Resource("CloudProfile").WithVersion("version"), "", admission.Update, &metav1.UpdateOptions{}, false, defaultUserInfo)
+
+				err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("should reject removal of kubernetes versions that are still in use by shoots", func() {
+				Expect(gardenCoreInformerFactory.Core().InternalVersion().Shoots().Informer().GetStore().Add(&shootOne)).To(Succeed())
+				Expect(gardenCoreInformerFactory.Core().InternalVersion().Shoots().Informer().GetStore().Add(&shootTwo)).To(Succeed())
+
+				cloudProfileNew := core.CloudProfile{
+					Spec: core.CloudProfileSpec{
+						Kubernetes: core.KubernetesSettings{
+							Versions: []core.ExpirableVersion{
+								{Version: "1.17.2"},
+							},
+						},
+					},
+				}
+
+				attrs := admission.NewAttributesRecord(&cloudProfileNew, &cloudProfile, core.Kind("CloudProfile").WithVersion("version"), "", cloudProfile.Name, core.Resource("CloudProfile").WithVersion("version"), "", admission.Update, &metav1.UpdateOptions{}, false, defaultUserInfo)
+
+				err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+				Expect(err).To(HaveOccurred())
```

Comment:
Can you please test, that expected error really occured?
```suggestion
				Expect(err).To(HaveOccurred())
				Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("unable to delete Kubernetes version %q from CloudProfile %q  - version is still in use by shoot %q/%q", "1.17.1", ...)))
```